src/lib/types.ts

```diff
@@ -34,6 +46,15 @@ export interface Config {
   readonly pillars: PillarConfig[];
 }
 
+/**
+ * Pillar scores for the three main evaluation pillars
+ */
+export interface PillarScores {
```

Comment:
i don't like this hardcoded. can you start with generating a "Pillars" enum from config.ts that looks like this (but generated):

```ts
const enum Pillar {
  MAINTENANCE: 'maintenance',
  QUALITY: 'quality',
  POPLARITY: 'popularity',
}
```

and then if you still need an inteface you can do something like:

```ts
interface PillarScores {
  [K in Pillar]: number;
}
```

might need some tinkering to get right, but try to do your best not to hardcode outside of config.ts